pypsa/linopf.py

```diff
@@ -1300,8 +1341,11 @@ def map_solution(c, attr):
             n.solutions.at[(c, attr), "pnl"] = False
             sol = variables.map(variables_sol)
             if predefined:
-                non_ext = n.df(c)[attr]
-                n.df(c)[attr + "_opt"] = sol.reindex(non_ext.index).fillna(non_ext)
+                if not (attr == "n_opt"):
+                    non_ext = n.df(c)[attr]
+                    n.df(c)[attr + "_opt"] = sol.reindex(non_ext.index).fillna(non_ext)
+                else:
+                    n.df(c).n_opt = sol
```

Comment:
This is delicate as it is clearly an ad-hoc fix and relatively difficult to generalize.
Currently, it may be accepted.
Alternatives may involve adding a new column in variables.csv to generalize and handle this exeptional case.

The main difference is the following: this if is on only for p_nom and e_nom that need the ending "*_opt", however with n_opt that is no more the case.
Do you have recommendations? Shall we keep it as is for the moment?